### DISCUSSION TITLE
change gap settings for specific workspaces

### DISCUSSION THREAD

**Discussion by chickenLemonade** [i3/i3#6092] [Q&A]
I'm trying to write a script that allows me to change the gap settings for a specific workspace. Does anybody know how to do this? The changing gaps section of the documentation only provides a way to do this for all workspaces, and running something like "i3-msg workspace 3 gaps inner 0" in the terminal does not work at all.

**Comment by KebabLord** [comment]
when using gaps, it either supports "current" or "all". It doesn't support workspace specific gap settings for now.
So for your script you can switch to the workspace first, and then set the gap for the active workspace:
`i3-msg workspace 1; i3-msg gaps inner current set 0`
